arangod/Agency/MoveShard.cpp

```diff
@@ -746,9 +752,24 @@ arangodb::Result MoveShard::abort() {
     return result;
   }
 
-  // Can now only be TODO or PENDING
+
+  // Can now only be TODO or PENDING. 
   if (_status == TODO) {
-    finish("", "", true, "job aborted");
+
+    // Do NOT remove, just cause it seems obvious!
+    // We're working off a snapshot.
+    // Make sure ToDo is still actually to be done
+    auto todoPrec = std::make_shared<Builder>();
+    { VPackArrayBuilder b(todoPrec.get());
+      { VPackObjectBuilder o(todoPrec.get()); } // nothing to declare
+      { VPackObjectBuilder path(todoPrec.get()); // expect jobs still to be sitting in ToDo
+        VPackValue(toDoPrefix + _jobId);
```

Comment:
Why does this compile in the first place? I think it needs to be `todoPrec->add(VPackValue(toDoPrefix + _jobId));`.